addons/mail/static/src/core/common/message.js

```diff
@@ -411,12 +411,15 @@ export class Message extends Component {
         if (ev.target.tagName === "A") {
             if (model && id) {
                 ev.preventDefault();
-                this.env.services.action.doAction({
+                await this.env.services.action.doAction({
                     type: "ir.actions.act_window",
                     res_model: model,
                     views: [[false, "form"]],
                     res_id: id,
                 });
+                if (!this.env.isSmall) {
+                    this.threadService.open(this.props.thread);
```

Comment:
For the auto-focus, Luc is okay to set the focus on the chat window as we don't know for sure why the user wanted to open the record:
- The users could open the record to update it. In that case, it's more suitable to put the focus on the form view.
- The users could open the record because they need additional info to answer the client. In that case, it's more suitable to put the focus on the chat window.

So, Luc said " the easiest implementation wins".

Reply:
I would still open thread without focus. I get that easiest implementation is sometimes preferred, but here this is annoying for click action to set the focus on a UI element that is not obviously related to the click at hand.
This can be especially problematic if the click shows a screen with a default focus, e.g. let's say it opens a list view and the search bar is automatically focused. Would be annoying that you have to explicitly click on search bar due to focus being on chat window instead.

Sometimes simpler is easier, but we should be cautious for this not resulting to poor UX.